.gitlab-ci.yml

```
@@ -0,0 +1,16 @@
+stages:
+  - build
+
+build:texlive:
+  stage: build
+  tags:
+    - docker
+  image: texlive/texlive
+  script:
+    - xelatex thesis
+    - '[ -f "thesis.aux" ] && biber thesis'
+    - xelatex thesis
+    - xelatex thesis
```

That's kind missing the point of this repository. You are encouraged to use the `Rakefile`, because it not only detects how often you need to rerun TeX and Biber (and in what order), it also has some niceties like extracting warnings and errors from the `.log` file.

Since `texlive/texlive` doesn't ship with Ruby, you need to install it first:

```suggestion
  before_script:
    - apt update
    - apt install --yes --no-install-recommends rake
  script:
    - rake
```

Of course, the proper way would be using an image with `rake` preinstalled.